### PR TITLE
Various changes to terraform configuration

### PR DESCRIPTION
**Pull request by relaxdiego**
Firstly, thank you for sharing this project. It's helped me a lot in getting updated with the latest in Kubernetes. I have a few proposed changes to help get a user up and running quickly. Specifically:

```
commit c85714e71f30784af8dbd43ba6904e3aba00b8bc

    Allow writing of variables to a *.tfvars file

    Specifying user variables such as the access_key, secret_key,
    default_keypair_public_key, and control_cidr in a *.tfvars file is
    more convenient since the user will not have to input it each time
    as well as have it in a file that's not committed to the repo.
```

```
commit 37ae447e5f623c2c2071643728eb2e3233a6a2f5

    Use Terraform's lookup feature for determining the AMI for a region

    Setting default AMIs for a given region reduces the steps needed to
    get up and running while also allowing the user to override it as
    needed.
```

### CODE REVIEW

**Comment by Bkbrawler04-zz** [comment]
+1

**Comment by relaxdiego** [comment]
@nicusX would love to know what you think of this :-)

File: .gitignore [hunk 1]
```diff
@@ -54,6 +54,8 @@ Temporary Items
 # Module directory
 .terraform/
 
+# Variables
+*.tfvars
```
Comment (nicusX) [inline .gitignore]: No need of this, as `terraform.tfvars` is already ignored in  ./terraform subdir.

**Comment by nicusX** [comment]
Hi guys,
thank you for the contribution. 

I totally agree about the list of AMI for all regions (I was a bit lazy about that :) ).
The variable defining the environment are already supposed to be in a `terraform.tfvars` file not to be committed and already ignored (thanks for the example file).

But I'm not keen on putting any secret in the tfvar file.

Environment configuration and secrets are totally different things, logically, and must remain separated.
In a real world project, configurations and secrets are stored in different places and injected in different ways (well... they should, as many people end up committing them in public repositories :P )

Personally, to speed up running this sample project, I had a bash file that exports the two secrets and add the pem file to ssh agent. 
So, not much burden running it, but separate from the `terraform.tfvar`

File: terraform/terraform.tfvars.example [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/* 
+   DO NOT EDIT terraform.tfvars.example! Copy it instead to 
+   terraform.tfvars and edit that file. 
+*/
+
+/* Required variables */
+access_key = "YOUR AWS ACCESS KEY HERE"
+secret_key = "YOUR AWS SECRET KEY HERE"
```
Comment (nicusX) [inline terraform/terraform.tfvars.example]: 👎 No secret mixed with environment configuration
(see my general comment)

File: terraform/variables.tf [hunk 3]
```diff
@@ -31,7 +31,8 @@ variable ansibleFilter {
   default = "Kubernetes01" # IF YOU CHANGE THIS YOU HAVE TO CHANGE instance_filters = tag:ansibleFilter=Kubernetes01 in ./ansible/hosts/ec2.ini
 }
 
-# Networking setup
+variable access_key {}
+variable secret_key {}
```
Comment (nicusX) [inline terraform/variables.tf]: Not required

**Comment by relaxdiego** [comment]
Thanks for the feedback, @nicusX! So if I understand it correctly, your bash file just does something like the following:

```
export AWS_ACCESS_KEY_ID="myaccesskeyid"
export AWS_SECRET_ACCESS_KEY="mysecret"
export TF_VAR_ACCESS_KEY=$ACCESS_KEY
export TF_VAR_SECRET_key=$SECRET_KEY
ssh-add /path/to/pem.file
```

Is that correct?

File: terraform/aws.tf [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # Retrieve AWS credentials from env variables AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY
 provider "aws" {
-  access_key = ""
-  secret_key = ""
+  access_key = "${var.access_key}"
+  secret_key = "${var.secret_key}"
```
Comment (nicusX) [inline terraform/aws.tf]: This is also not required

**Comment by nicusX** [comment]
My "secret" script is as simple as the following:

```
export AWS_ACCESS_KEY_ID="FOO"
export AWS_SECRET_ACCESS_KEY="BAR"
ssh-add /path/mykey.pem
```

As documented https://www.terraform.io/docs/providers/aws/  Terraform automatically reads `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` environment variable.
No need to define them as variables.

**Comment by relaxdiego** [comment]
Ah! Thanks for clarifying. Clearly I am new to Terraform so I appreciate this. :-) I'll update this PR in a few minutes.

**Review by nicusX — APPROVED** [review]

**Pull request merged**